designs/2019-config-tester/README.md

```diff
@@ -0,0 +1,108 @@
+- Start Date: 2019-06-14
+- RFC PR: https://github.com/eslint/rfcs/pull/27
+- Authors: Toru Nagashima &lt;https://github.com/mysticatea&gt;
+
+# Config Tester
+
+## Summary
+
+Providing `ConfigTester` API that tests shareable configs and plugin's preset configs.
+
+## Motivation
+
+Currently, we don't provide stuff to test shareable configs. This means the community doesn't have an easy way to check if their config is good or not.
+
+- Is the config structure valid?
+- Does every rule exist?
```

Comment:
This should check rules that exist in previous versions of eslint too - ideally that match the eslint peer dep in the config/plugin - for back compat, a rule might be listed that works in eslint 2, 3, 4, 5, or 6, and that shouldn't cause an error if that eslint version is supported.

Reply:
Thank you for your feedback.

Are you talking about testing the latest config with old version's `ConfigTester` in the future?

Reply:
I'm talking about using ConfigTester to test a version of eslint-config-airbnb that supports both eslint v4 and v5 and v6 at the same time, and has rules in it that aren't in all three versions, aren't in v6, etc.

Reply:
I see. Does `ignoreDisabledNonexistenceRules`-like option work for you?

Reply:
I guess what I'd prefer is that I can always use the latest version of ConfigTester, but that it will contain the knowledge to validate all current and previous eslint versions.

Reply:
I think it's really hard because plugins don't have a uniformed update history. Similarly we do tests with multiple versions of Node.js, CI with multiple versions of the underlying is relatively easy.